benchmark/framework/aea_test_wrapper.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This test module contains AEA/AEABuilder wrapper to make performance tests easy."""
+from threading import Thread
+from typing import Dict, List, Optional, Tuple, Type, Union
+
+from benchmark.framework.fake_connection import FakeConnection
+
+from aea.aea import AEA
+from aea.aea_builder import AEABuilder
+from aea.configurations.base import ProtocolId, SkillConfig
+from aea.configurations.components import Component
+from aea.crypto.fetchai import FETCHAI
+from aea.mail.base import Envelope
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.skills.base import Handler, Skill, SkillContext
+
+
+class AEATestWrapper:
+    """A testing wrapper to run and control an agent."""
+
+    def __init__(self, name: str = "my_aea", skills: List[dict] = None):
+        """Make an agency with optional name and skills."""
+        self.skills = skills or []
+        self.name = name
+
+        self.aea = self.make_aea(
+            self.name, [self.make_skill(**skill) for skill in self.skills]
+        )
+
+    def make_aea(self, name: str = "my_eae", components: List[Component] = None) -> AEA:
+        """
+        Create AEA from name and already loaded components.
+
+        :return: AEA
+        """
+        components = components or []
+        builder = AEABuilder()
+
+        builder.set_name(self.name)
+        builder.add_private_key(FETCHAI, "")
+
+        for component in components:
+            builder._resources.add_component(component)
+
+        aea = builder.build()
+        return aea
+
+    def make_skill(
+        self,
+        config: SkillConfig = None,
+        context: SkillContext = None,
+        handlers: Optional[Dict[str, Type[Handler]]] = None,
+    ) -> Skill:
+        """
+        Make a skill from optional config, context, handlers dict.
+
```

Comment:
missing params in docstring

Reply:
note, the signature should also include `behaviours: Optional[Dict[str, Type[Behaviour]]] = None, models: Optional[Dict[str, Type[Model]]] = None`

Reply:
docstrings are fixed
@DavidMinarsch 
probably this part should be unified with the programmatical way of agent's creation.
for the moment it's just a helper for the current test cases and it's not assumed it should cover all the possible cases of agent's creation